cdap-api/src/main/java/co/cask/cdap/api/templates/plugins/PluginProperties.java

```diff
@@ -80,7 +80,7 @@ public Builder add(String key, String value) {
      * Creates a new instance of {@link PluginProperties} with the properties added to this builder prior to this call.
      */
     public PluginProperties build() {
-      return new PluginProperties(Collections.unmodifiableMap(new HashMap<>(properties)));
+      return new PluginProperties(new HashMap<>(properties));
```

Comment:
Why the change? The `PluginProperties` doesn't do any defensive copying and relaying on the fact that the builder gives it an immutable map (which will get returned through the public `getProperties` method).